collect_app/src/test/java/org/odk/collect/android/formentry/RecordingHandlerTest.java

```diff
@@ -42,6 +48,33 @@ public class RecordingHandlerTest {
     @Before
     public void setup() {
         recordingHandler = new RecordingHandler(questionMediaManager, new FakeLifecycleOwner(), mock(AudioRecorder.class), amrAppender, m4aAppender);
+        doReturn(formDef).when(formController).getFormDef();
+    }
+
+    @Test
+    public void whenBackgroundRecordingM4A_andThereIsNoRecordingAlreadySavedForReference_savesNewAnswer() throws Exception {
+        TreeReference treeReference = new TreeReference();
+
+        File newRecording = File.createTempFile("new", ".m4a");
+        RecordingSession recordingSession = new RecordingSession(new HashSet<TreeReference>() {
+            {
+                add(treeReference);
+            }
+        }, newRecording, 0, 0, false);
+
+        recordingHandler.handle(formController, recordingSession, success -> {
+        });
+
+        ArgumentCaptor<StringData> dataAc = ArgumentCaptor.forClass(StringData.class);
+        ArgumentCaptor<TreeReference> refAc = ArgumentCaptor.forClass(TreeReference.class);
+        ArgumentCaptor<Boolean> midSurveyAc = ArgumentCaptor.forClass(Boolean.class);
+
+        verify(formDef).setValue(dataAc.capture(), refAc.capture(), midSurveyAc.capture());
```

Comment:
I think you can just pass the expected values here or use `eq` rather than needing `ArgumentCaptor`? Am I missing something?

Reply:
Right, fixed.